chpl/chpl-service/src/main/java/gov/healthit/chpl/validation/listing/reviewer/edition2015/InvalidCriteriaCombinationReviewer.java

```diff
@@ -0,0 +1,58 @@
+package gov.healthit.chpl.validation.listing.reviewer.edition2015;
+
+import java.util.Optional;
+
+import org.ff4j.FF4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import gov.healthit.chpl.FeatureList;
+import gov.healthit.chpl.domain.CertificationResult;
+import gov.healthit.chpl.domain.CertifiedProductSearchDetails;
+import gov.healthit.chpl.util.ErrorMessageUtil;
+import gov.healthit.chpl.validation.listing.reviewer.Reviewer;
+
+@Component
+public class InvalidCriteriaCombinationReviewer implements Reviewer {
+    private static final String B6 = "170.315 (b)(6)";
+    private static final String B10 = "170.315 (b)(10)";
+
+    private static final String G8 = "170.315 (g)(8)";
+    private static final String G10 = "170.315 (g)(10)";
+
+    private ErrorMessageUtil msgUtil;
+    private FF4j ff4j;
+
+    @Autowired
+    public InvalidCriteriaCombinationReviewer(ErrorMessageUtil msgUtil, FF4j ff4j) {
+        this.msgUtil = msgUtil;
+        this.ff4j = ff4j;
+    }
+
+    @Override
+    public void review(CertifiedProductSearchDetails listing) {
+        if (ff4j.check(FeatureList.EFFECTIVE_RULE_DATE)) {
+            checkForInvalidCriteriaCombination(listing, B6, B10);
+            checkForInvalidCriteriaCombination(listing, G8, G10);
```

Comment:
I'm going to approve this as written, but if you have time, we might use this same reviewer to make sure that the (for instance) (b)(1) and (b)(1)-Cures Update criteria can't be attested to at the same time. The numbers are the same for both; it's only the title that's different. If it is reasonably easy to add that check to this reviewer that'll make our lives easier going forward, but if not, NBD; we can do it later